File: src/main/java/tech/zolhungaj/amqapi/servercommands/gameroom/game/PlayerRejoin.java
package tech.zolhungaj.amqapi.servercommands.gameroom.game;

import com.squareup.moshi.Json;
import tech.zolhungaj.amqapi.servercommands.Command;
import tech.zolhungaj.amqapi.servercommands.CommandTypeOld;

public record PlayerRejoin(
        int gamePlayerId,
        @Json(name = "name")
        String playerName
) implements Command {
    @Override
    public String commandName() {
        return CommandTypeOld.PLAYER_REJOIN.commandName;
    }
}

File: src/main/java/tech/zolhungaj/amqapi/servercommands/gameroom/game/PlayerRejoin.kt
package tech.zolhungaj.amqapi.servercommands.gameroom.game;

import com.squareup.moshi.Json;
import tech.zolhungaj.amqapi.servercommands.CommandType;

@CommandType("Rejoining Player")
public record PlayerRejoin(
        int gamePlayerId,
        @Json(name = "name")
        String playerName
){}
